### PR TITLE
Renter encryption

### PR DESCRIPTION
**Pull request by DavidVorick**
This builds on the magnet links pull request.

I added encryption to the renter. I also found the bug that was preventing the host from getting paid for storing files.

As far as my limited testing can tell, everything is good to go. This is the best Sia has ever worked to the extent of my knowledge, and it should not be confusing to use any more.

### CODE REVIEW

File: modules/host/update.go [hunk 1]
```diff
@@ -60,17 +60,16 @@ func (h *Host) ReceiveConsensusSetUpdate(revertedBlocks []types.Block, appliedBl
 	lockID := h.mu.Lock()
 	defer h.mu.Unlock(lockID)
 
+	for _ = range revertedBlocks {
+		h.blockHeight--
+	}
```
Comment (lukechampine) [inline modules/host/update.go]: maybe `h.blockHeight - len(revertedBlocks)`?

File: modules/renter/download.go [hunk 2]
```diff
@@ -112,9 +113,18 @@ func (d *Download) start() {
 		for _, piece := range d.pieces {
 			downloadErr := d.downloadPiece(piece)
 			if downloadErr == nil {
-				d.complete = true
-				d.file.Close()
-				return
+				// Decrypt the file.
+				d.file.Seek(0, 0)
+				cryptBytes, err := ioutil.ReadAll(d.file)
```
Comment (lukechampine) [inline modules/renter/download.go]: storing the entire file in memory?

File: modules/renter/persist.go [hunk 3]
```diff
@@ -184,6 +184,16 @@ func (r *Renter) loadSharedFile(reader io.Reader) error {
...
 	for i := range rsf.Files {
+		for {
+			_, exists := r.files[rsf.Files[i].Name]
+			if !exists {
+				break
+			}
+			if len(rsf.Files[i].Name) > 50 {
+				break
+			}
+			rsf.Files[i].Name += "_"
```
Comment (lukechampine) [inline modules/renter/persist.go]: how about
```go
for i := range rsf.Files
    if _, exists := r.files[rsf.Files[i].Name]; !exists {
        continue
    }
    // if a file with this name already exists, add a duplicate number
    dupCount := 1
    for {
        name := rsf.Files[i].Name + strconv.Itoa(dupCount)
        _, exists := r.files[name]
        if !exists {
            rsf.Files[i].Name = name
            break
        }
        dupCount++
    }
}
```

File: modules/host/update.go (hunk 1 above)
Comment (DavidVorick) [inline modules/host/update.go]: I don't think there's anything else to do in the loop

File: modules/renter/download.go (hunk 2 above)
Comment (DavidVorick) [inline modules/renter/download.go]: this is a short term solution. The longer term solution involves loading about 10MB at a time and leveraging micro-diffs to keep all of the memory requirements sane.

File: modules/renter/persist.go (hunk 3 above)
Comment (DavidVorick) [inline modules/renter/persist.go]: ok

File: modules/renter/download.go (hunk 2 above)
Comment (lukechampine) [inline modules/renter/download.go]: yeah I figured, just checking

**Comment by lukechampine** [comment]
Remember to update the .gitignore and Makefile

**Comment by DavidVorick** [comment]
There aren't any tests for the changes, but I tested it manually and the file renaming works as expected (loading the same file repeatedly goes: file2 -> file2_1 -> file2_2 -> file2_3)

**Pull request merged**